rust/perspective-viewer/src/rust/components/column_settings_sidebar/style_tab.rs

```diff
@@ -0,0 +1,224 @@
+// ┃ █▀▀▀▀▀ █▀▀▀▀▀ █▀██▀▀ ▄▄▄▄▄ █ ▄▄▄▄▄█ ▄▄▄▄▄█ ████████▌▐███ █████▄   █ ▄▄▄▄▄ ┃
+// ┃ █      ██████ █  ▀█▄       █ ██████      █      ███▌▐███ ███████▄ █       ┃
+// ┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┫
+// ┃ Copyright (c) 2017, the Perspective Authors.                              ┃
+// ┃ ╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌ ┃
+// ┃ This file is part of the Perspective library, distributed under the terms ┃
+// ┃ of the [Apache License 2.0](https://www.apache.org/licenses/LICENSE-2.0). ┃
+// ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+
+use std::collections::HashMap;
+use std::fmt::Debug;
+
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+use wasm_bindgen::{JsCast, JsValue};
+use yew::{function_component, html, Callback, Html, Properties};
+
+use crate::components::datetime_column_style::DatetimeColumnStyle;
+use crate::components::number_column_style::NumberColumnStyle;
+use crate::components::string_column_style::StringColumnStyle;
+use crate::components::style::LocalStyle;
+use crate::config::Type;
+use crate::presentation::Presentation;
+use crate::renderer::Renderer;
+use crate::session::Session;
+use crate::utils::JsValueSerdeExt;
+use crate::{clone, css, html_template};
+
+#[derive(Clone, PartialEq, Properties)]
+pub struct StyleTabProps {
+    pub session: Session,
+    pub renderer: Renderer,
+    pub presentation: Presentation,
+    pub column_name: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct DefaultConfig {
+    string: serde_json::Value,
+    datetime: serde_json::Value,
+    date: serde_json::Value,
+    integer: serde_json::Value,
+    float: serde_json::Value,
+    bool: serde_json::Value,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct Config {
+    columns: HashMap<String, serde_json::Value>,
+    default_config: Option<DefaultConfig>,
+}
+
+/// This function sends the config to the plugin using its `restore` method
+fn send_config<T: Serialize + Debug>(
+    renderer: Renderer,
+    presentation: Presentation,
+    view: JsValue,
+    column_name: String,
+    column_config: T,
+) {
+    let current_config = get_config(&renderer);
+    let elem = renderer.get_active_plugin().unwrap();
+    if let Some(mut current_config) = current_config {
+        current_config
+            .columns
+            .insert(column_name, serde_json::to_value(column_config).unwrap());
+        let js_config = JsValue::from_serde_ext(&current_config).unwrap();
+        elem.restore(&js_config);
+        // we don't need to block yew's rendering to restyle the element,
+        // nor do we need any returned result, nor do we care if it fails.
+        wasm_bindgen_futures::spawn_local(async move {
+            let view = view.unchecked_into();
+            elem.update(&view, None, None, false)
+                .await
+                .expect("Unable to call update!");
```

Comment:
While we don't care if this fails, generally panicking in Yew is bad, as it can leave the app in an inconsistent state. `ApiFuture::spawn` expects a `ApiResult` return type which is compatible with update and correctly routes this failure to the `Promise.reject()` method, both reporting the failure in the console properly _and_ avoiding panic.